web3.js/test/esm-compat.mjs

```diff
@@ -0,0 +1,4 @@
+// This tests if the package is configured correctly and can be imported in Node versions that support ESM.
+import * as web3 from '@solana/web3.js';
+
+console.log(web3);
```

Comment:
While this is a good test when looking at it, I don't think it will fail the CI if it fails. Try using mocha framework `describe` `it` like the other tests, so that it fails when we test through mocha.

Reply:
It should fail CI because Node will just throw an exception if it cannot import the module, I can make it a mocha test but it should fail in exactly the same way, no `describe` or `it` blocks will ever be reached if that happens

Reply:
Using a dynamic import might work, ok let me try that...

Reply:
I see you added a different step in the CI. I think that should cover it. I think something needs to be updated unrelated to the PR to fix the build issue. Will wait for another reviewer to get their eyes on this.

Reply:
Makes sense, I'll update

Reply:
Had to remove the test as it turns out there's no way to test ESM compatibility without either changing the build files to `.mjs` or switching  `"type"` to `"module"` in `package.json`. I'm happy to do it if there's willingness from the authors to accept a change that potentially might break compatibility with some legacy bundlers